build.gradle.kts

```diff
@@ -616,11 +694,101 @@ allprojects {
+     */
+
+    val transitiveConfigurations = setOf(
+        "androidTestDebugImplementation",
+        "androidTestDebugImplementationDependenciesMetadata",
+        "androidTestImplementation",
+        "androidTestImplementationDependenciesMetadata",
+        "androidTestReleaseImplementation",
+        "androidTestReleaseImplementationDependenciesMetadata",
+        "annotationProcessor",
+        "debugAndroidTestCompilationImplementation",
+        "debugAndroidTestImplementation",
+        "debugAndroidTestImplementationDependenciesMetadata",
+        "debugAnnotationProcessor",
+        "debugAnnotationProcessorClasspath",
+        "debugUnitTestCompilationImplementation",
+        "debugUnitTestImplementation",
+        "debugUnitTestImplementationDependenciesMetadata",
+        "kotlinBuildToolsApiClasspath",
+        "kotlinCompilerClasspath",
+        "kotlinCompilerPluginClasspath",
+        "kotlinCompilerPluginClasspathDebug",
+        "kotlinCompilerPluginClasspathDebugAndroidTest",
+        "kotlinCompilerPluginClasspathDebugUnitTest",
+        "kotlinCompilerPluginClasspathMain",
+        "kotlinCompilerPluginClasspathRelease",
+        "kotlinCompilerPluginClasspathReleaseUnitTest",
+        "kotlinCompilerPluginClasspathTest",
+        "kotlinKlibCommonizerClasspath",
+        "kotlinNativeCompilerPluginClasspath",
+        "kotlinScriptDef",
+        "kotlinScriptDefExtensions",
+        "mainSourceElements",
+        "releaseAnnotationProcessor",
+        "releaseAnnotationProcessorClasspath",
+        "releaseUnitTestCompilationImplementation",
+        "releaseUnitTestImplementation",
+        "releaseUnitTestImplementationDependenciesMetadata",
+        "testDebugImplementation",
+        "testDebugImplementationDependenciesMetadata",
+        "testFixturesDebugImplementation",
+        "testFixturesDebugImplementationDependenciesMetadata",
+        "testFixturesImplementation",
+        "testFixturesImplementationDependenciesMetadata",
+        "testFixturesReleaseImplementation",
+        "testFixturesReleaseImplementationDependenciesMetadata",
+        "testImplementation",
+        "testImplementationDependenciesMetadata",
+        "testReleaseImplementation",
+        "testReleaseImplementationDependenciesMetadata",
+    )
+
+    /*
+     * Write the set of available configurations to files, for debugging purposes. Plugins can
+     * add new configurations at any time, and so it's nice to have a list of the available
+     * configurations visible.
+     */
+
+    val configurationsActual = mutableSetOf<String>()
+    afterEvaluate {
+        configurations.all {
+            configurationsActual.add(this.name)
+        }
+        File("configurations.txt").writeText(configurationsActual.joinToString("\n"))
+    }
+
+    afterEvaluate {
+        configurations.all {
+            isTransitive = transitiveConfigurations.contains(name)
+            // resolutionStrategy.failOnVersionConflict()
```

Comment:
Was this comment left here on purpose?

Reply:
It was copied in from the `android-core` repository script. It's fine to remove it, but I left it in as I found myself repeatedly turning it on and off during development. It's something I'd like to leave permanently on, but we're not quite there yet.

Reply:
Sure, there's no problem in keeping it. I was just asking